game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AirBattleOffenseCombatValue.java

```diff
@@ -0,0 +1,93 @@
+package games.strategy.triplea.delegate.power.calculator;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.TerritoryEffect;
+import games.strategy.engine.data.Unit;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Value;
+import org.triplea.java.collections.IntegerMap;
+
+/** Calculates offense strength and roll for Air Battle dice */
+@Builder
+@Value
+@Getter(AccessLevel.NONE)
+class AirBattleOffenseCombatValue implements CombatValue {
+
+  @Getter(onMethod = @__({@Override}))
+  @NonNull
+  GameData gameData;
```

Comment:
Having a reference to a game data is a bit of a step backwards. It's a god-class and the fix for it is to avoid adding references to it. 

What if instead of passing the 'gameData' reference to the builder of this class, we push the responsibility getting the dice sides to the caller? That would also make this more of a value class as well where it really just holds values.

Reply:
I'll work on that after this PR.